.CMake/alg_support.cmake

```diff
@@ -2,6 +2,48 @@
 
 include(CMakeDependentOption)
 
+# Switch off all algs except for those passed in the alglist
+function(filter_algs alglist)
+  # Set every OQS_ENABLE_* variable =OFF unless one of the following conditions holds:
+  #  1. the switch for one of the requested minimal build algorithm's family, e.g OQS_ENABLE_KEM_KYBER
+  #  2. the switch for one of the requested algorithms, e.g. OQS_ENABLE_KEM_kyber_768.
+  #  3. the switch for platform-specific ("_aesni" or "_avx2") implementation of
+  #     one of the requested algorithms, e.g. OQS_ENABLE_KEM_kyber_768_avx2.
+
+  get_cmake_property(_vars VARIABLES)
+  foreach (_var ${_vars})
+      if(_var MATCHES "^OQS_ENABLE_..._" AND NOT _var MATCHES "_AVAILABLE$")
+          set(${_var} OFF PARENT_SCOPE)
+          # Case 1, family name
+	  foreach (_alg ${ARGV0})
+             string(TOUPPER ${_alg} upalg)
+	     if("OQS_ENABLE_${upalg}i" MATCHES "^${_var}")
+                 set(${_var} ON PARENT_SCOPE)
+             endif()
+          endforeach()
+          # Case 2, exact match
```

Comment:
Rather than this sort of finicky parsing, wouldn't it be preferable to have additional `{nist-r4|std}.cmake` files where we have a bunch of `option(OQS_ENABLE_{KEM|SIG}_{ALG} "Enable {ALG} algorithm family" ON)` lines (these could be auto-generated).

Reply:
Why create more code if existing code does the job? Besides, this code until now successfully drove our "OQS_MINIMAL_BUILD" logic, so arguably is proven to work OK.